mod/caching/set/type_plus_right/wikirate_company/wikirate_topic.rb

```diff
@@ -1,33 +1,47 @@
 include Card::CachedCount
-#{
-#   "type":"topic",
-#   "referred_to_by":{
-#     "or":{
-#       "left":{
-#           "type":["in","Note","Source"],
-#           "right_plus":["company",{"refer_to":"_1"}]
-#       },
-#       "and":{
-#         "type":"Metric",
-#         "right_plus":["_1",{}]
-#        }
-#     },
-#     "right":"topic"
-#   }
-# }
+
+def related_topic_from_source_or_note
+  Card.search(
+    type: 'Topic',
+    referred_to_by: {
+      left: {
+        type: %w(in Note Source),
+        right_plus: ['company', refer_to: cardname.left]
+      },
+      right: 'topic'
+    },
+    return: 'id')
+end
+
+def related_topic_from_metric
+  Card.search(
+    type: 'Topic',
+    referred_to_by: {
+      left: { type: 'Metric', right_plus: cardname.left },
+      right: 'topic'
+    },
+    return: 'id')
+end
+
+# get all metric values
+def calculate_count
+  (related_topic_from_source_or_note + related_topic_from_metric).uniq.size
+end
 
 # recount topics associated with a company whenever <source>+company is edited
 ensure_set { TypePlusRight::Source::WikiRateCompany }
 recount_trigger TypePlusRight::Source::WikirateCompany do |changed_card|
-  changed_card.item_names.map do |company_name|
+  names = Card::CachedCount.pointer_card_changed_card_names(changed_card)
+  names.map do |company_name|
```

Comment:
I like the idea here.

Wouldn't it make more sense to put the change detection code in type/pointer and call it with something like "changed_card.item_names_updated_in_last_action"?